packages/@webex/plugin-meetings/src/multistream/sendSlotManager.ts

```diff
@@ -0,0 +1,165 @@
+   * @param {boolean} active This is optional boolean to set the active state of the sendSlot. Default is true
+   * @returns {SendSlot} slot The created sendSlot
+   */
+  public createSlot(
+    mediaConnection: MultistreamRoapMediaConnection,
+    mediaType: MediaType,
+    active = true
+  ): SendSlot {
+    if (this.slots.has(mediaType)) {
+      throw new Error(`Slot for ${mediaType} already exists`);
+    }
+
+    const slot: SendSlot = mediaConnection.createSendSlot(mediaType, active);
+
+    this.slots.set(mediaType, slot);
+
+    this.LoggerProxy.logger.info(
+      `SendSlotsManager->createSlot#Created slot for ${mediaType} with active ${active}`
+    );
+
+    return slot;
+  }
+
+  /**
+   * This method is used to retrieve the sendSlot for the given mediaType
+   * @param {MediaType} mediaType of which the slot needs to be retrieved
+   * @returns {SendSlot}
+   */
+  public getSlot(mediaType: MediaType): SendSlot {
+    const slot = this.slots.get(mediaType);
+
+    if (!slot) {
+      throw new Error(`Slot for ${mediaType} does not exist`);
+    }
+
+    return slot;
+  }
+
+  /**
+   * This method publishes the given stream to the sendSlot for the given mediaType
+   * @param {MediaType} mediaType MediaType of the sendSlot to which a stream needs to be published (AUDIO_MAIN/VIDEO_MAIN/AUDIO_SLIDES/VIDEO_SLIDES)
+   * @param {LocalStream} stream LocalStream to be published
+   * @returns {Promise<void>}
+   */
+  public async publishStream(mediaType: MediaType, stream: LocalStream): Promise<void> {
+    const slot = this.slots.get(mediaType);
+
+    if (!slot) {
+      throw new Error(`Slot for ${mediaType} does not exist`);
+    }
+
+    await slot.publishStream(stream);
+
+    this.LoggerProxy.logger.info(
+      `SendSlotsManager->publishStream#Published stream for ${mediaType} and stream ${stream}`
+    );
+  }
+
+  /**
+   * This method unpublishes the stream from the sendSlot of the given mediaType
+   * @param {MediaType} mediaType MediaType of the sendSlot from which a stream needs to be unpublished (AUDIO_MAIN/VIDEO_MAIN/AUDIO_SLIDES/VIDEO_SLIDES)
+   * @returns {Promise<void>}
+   */
+  public async unpublishStream(mediaType: MediaType): Promise<void> {
+    const slot = this.slots.get(mediaType);
+
+    if (!slot) {
+      throw new Error(`Slot for ${mediaType} does not exist`);
+    }
+
+    await slot.unpublishStream();
+
+    this.LoggerProxy.logger.info(
+      `SendSlotsManager->unpublishStream#Unpublished stream for ${mediaType}`
+    );
+  }
+
+  /**
+   * This method is used to toggle the active state of the sendSlot for the given mediaType
+   * @param {MediaType} mediaType The MediaType of the sendSlot for which the active state needs to be toggled (AUDIO_MAIN/VIDEO_MAIN/AUDIO_SLIDES/VIDEO_SLIDES)
+   * @param {boolean} active The optional boolean to set the active state of the sendSlot
+   * @returns {void}
+   */
+  public toggleActive(mediaType: MediaType, active?: boolean): void {
```

Comment:
Perhaps this should be setActive rather than toggleActive? Toggle generally doesn't require an input of the boolean.

Reply:
That's correct but wanted to have it both ways. If you notice, the active boolean is optional, if they explicitly provide a boolean for it, we'll set it, if not we'll toggle by default. Quoting my comment from [an earlier review](https://github.com/mkesavan13/webex-js-sdk/pull/1#discussion_r1301988887) of Bryce: 

> The aim of the method is to toggleActive state either based on
> - the current active state (which is why if the active boolean is undefined, we will invert)
> - the given active parameter (It's optional. If it's given, we won't worry about the current state, rather set it to the >given value)

Reply:
Wondering if it would be better to just have it one way or another instead of allowing both "set" and "toggle" here, considering both me and @sreenara were both initially confused by it. (If we were to choose one, I think I'd rather have it be "set" instead of "toggle".)

At the very least, I think the JSDoc should explain this more explicitly (i.e. if `active` is omitted, the API will treat this function as a "toggle" rather than a "set").

Reply:
Okay. It sure makes sense to keep it that way. Unlike mute, no on is going to use this method as toggle. 

(I made a few changes for updateMedia today that gave this thought) Will go ahead and change it to `setActive`